ci/common/build.bash

```diff
@@ -0,0 +1,89 @@
+#! /usr/bin/env bash
+
+# Copyright (c) 2018-2020 NVIDIA Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Released under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+
+################################################################################
+# Thrust and CUB build script for gpuCI
+################################################################################
+
+set -e
+
+# Logger function for build status output
+function logger() {
+  echo -e "\n>>>> ${@}\n"
+}
+
+################################################################################
+# VARIABLES - Set up bash and environmental variables.
+################################################################################
+
+# Get the variables the Docker container set up for us: ${CXX}, ${CUDACXX}, etc.
+source /etc/cccl.bashrc
+
+# Set path and build parallel level
+export PATH=/usr/local/cuda/bin:${PATH}
+
+# Set home to the job's workspace.
+export HOME=${WORKSPACE}
+
+# Switch to project root; also root of repo checkout.
+cd ${WORKSPACE}
+
+# If it's a nightly build, append current YYMMDD to version.
+if [ "${BUILD_MODE}" == "branch" ]; then
+  export VERSION_SUFFIX=`date +%y%m%d`
+fi
+
+# The Docker image sets up `${CXX}` and `${CUDACXX}`.
+CMAKE_FLAGS="-DCMAKE_CXX_COMPILER=${CXX} -DCMAKE_CUDA_COMPILER=${CUDACXX}"
+
+# If it's a nightly build, build all configurations.
+if [ "${BUILD_MODE}" == "branch" ]; then
+  CMAKE_FLAGS="${CMAKE_FLAGS} -DTHRUST_MULTICONFIG_WORKLOAD=FULL"
```

Comment:
`THRUST_MULTICONFIG_WORKLOAD=FULL` has no effect unless multiconfig builds are enabled. We should also explicitly specify the dialects and systems we want to test and tell CMake to include CUB's tests.

I've documented Thrust's CMake options [here](https://github.com/NVIDIA/thrust/blob/main/CONTRIBUTING.md#cmake-options). See [this comment](https://github.com/NVIDIA/thrust/issues/1164#issuecomment-703163439) for an example multiconfig setup.